inc/commonitilobject.class.php

```diff
@@ -7101,4 +7101,21 @@ public function isStatusComputationBlocked(array $input) {
       return array_key_exists('_do_not_compute_status', $input)
          && $input['_do_not_compute_status'];
    }
+
+   /**
+    * Check if this item is new
+    *
+    * @return bool|null null if we can't check the status
+    */
+   protected function isNew() {
+      if (isset($this->input['status'])) {
+         $status = $this->input['status'];
+      } else if (isset($this->fields['status'])) {
+         $status = $this->fields['status'];
+      } else {
+         return null;
```

Comment:
This is uncommon and seems dangerous, as most of developper will check returned value as a boolean and `null` will be considered as `false`.
Maybe this method should try to load fields from DB if `!$this->isNewItem()`, and fallback to `true` if field is not set and `$this->isNewItem()`.

Reply:
I though of this `null == false` problem too but I'm not sure returning true is any better.

The problem is still that we use the same class for entities and repositories...
In the case of a repository, `true` or `false` doesn't make any sense (no object are loaded, how can we say whether or not the status is new ?) so I thought `null` was more fitting despite the loose comparisons problems.

Maybe we should throw a `LogicException` instead to prevent running this method if no objects are loaded ?

Reply:
`LogicException` may be the fastest and most robust solution for now.